chia/_tests/environments/wallet.py

```diff
@@ -345,7 +345,9 @@ async def process_pending_states(self, state_transitions: List[WalletStateTransi
             try:
                 await self.full_node.check_transactions_confirmed(env.wallet_state_manager, txs)
             except TimeoutError:  # pragma: no cover
-                unconfirmed: List[TransactionRecord] = await env.wallet_state_manager.tx_store.get_all_unconfirmed()
+                unconfirmed: List[LightTransactionRecord] = (
+                    await env.wallet_state_manager.tx_store.get_all_unconfirmed()
+                )
                 raise TimeoutError(
                     f"ENV-{i} TXs not confirmed: {[tx.to_json_dict() for tx in unconfirmed if tx in txs]}"
```

Comment:
for example; we ask for all unconfirmed here, but then we immediately filter to only include the ones in `txs`